### PR TITLE
store: cache suggested currency from the store

### PR DESCRIPTION
**Pull request by pete-woods**
No CLI / REST support yet

### CODE REVIEW

File: store/snap_remote_repo.go [hunk 1]
```diff
@@ -241,6 +245,21 @@ func (s *SnapUbuntuStoreRepository) configureStoreReq(req *http.Request, accept
 	s.applyUbuntuStoreHeaders(req, accept)
 }
 
+// read all the available metadata from the store response and cache
+func (s *SnapUbuntuStoreRepository) readStoreResponse(resp *http.Response) {
+	suggestedCurrency := resp.Header.Get("X-Suggested-Currency")
+	if suggestedCurrency == "" {
```
Comment (niemeyer) [inline store/snap_remote_repo.go]: That logic means we'll replace a known currency by a default value. I think we should do this instead:
```
suggestedCurrency := ... from header ...

s.mu.Lock()
defer s.mu.Unlock()
if suggestedCurrency != "" {
        s.suggestedCurrency = suggestedCurrency
}
```

We'll handle defaults when getting it.

Comment (niemeyer) [inline store/snap_remote_repo.go]: (made locking slightly more reasonable)

File: store/snap_remote_repo.go [hunk 2]
```diff
@@ -241,6 +245,21 @@ func (s *SnapUbuntuStoreRepository) configureStoreReq(req *http.Request, accept
 	s.applyUbuntuStoreHeaders(req, accept)
 }
 
+// read all the available metadata from the store response and cache
+func (s *SnapUbuntuStoreRepository) readStoreResponse(resp *http.Response) {
```
Comment (niemeyer) [inline store/snap_remote_repo.go]: "reading a response" has a specific meaning which isn't intended here.

Let's please name this `checkStoreResponse`.

File: store/snap_remote_repo.go [hunk 3]
```diff
@@ -474,3 +499,11 @@ func (s *SnapUbuntuStoreRepository) Assertion(assertType *asserts.AssertionType,
 	dec := asserts.NewDecoder(resp.Body)
 	return dec.Decode()
 }
+
+// SuggestedCurrency retrieves the cached value for the store's suggested currency
+func (s *SnapUbuntuStoreRepository) SuggestedCurrency() string {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	return s.suggestedCurrency
```
Comment (niemeyer) [inline store/snap_remote_repo.go]:
```
if s.suggestedCurrency == "" {
        return "USD"
}
return s.suggestedCurrency
```

**Comment by niemeyer** [comment]
LGTM

**Comment by chipaca** [comment]
:+1:

**Pull request merged**